lib/studies/reply/mail_card_preview.dart

```diff
@@ -26,72 +38,95 @@ class MailPreviewCard extends StatelessWidget {
+                              if (isDesktop)
+                                Row(
+                                  children: const [
+                                    ImageIcon(
+                                      AssetImage(
+                                        '$_iconAssetLocation/twotone_star.png',
+                                        package: _assetsPackage,
+                                      ),
+                                      color: ReplyColors.blue600,
+                                    ),
+                                    SizedBox(width: 20),
+                                    ImageIcon(
+                                      AssetImage(
+                                        '$_iconAssetLocation/twotone_delete.png',
+                                        package: _assetsPackage,
+                                      ),
+                                      color: ReplyColors.blue600,
+                                    ),
+                                    SizedBox(width: 20),
+                                    Icon(Icons.more_vert,
+                                        color: ReplyColors.blue700),
+                                  ],
+                                ),
+                              const SizedBox(width: 20),
+                              Transform.translate(
+                                offset: Offset(0, isDesktop ? -16 : 8),
+                                child: ClipOval(
+                                  child: Image.asset(
+                                    email.avatar,
+                                    package: _assetsPackage,
+                                    height: 36,
+                                    width: 36,
+                                  ),
+                                ),
+                              ),
+                            ],
                           ),
-                          SizedBox(width: 20),
-                          Icon(Icons.more_vert, color: ReplyColors.blue700),
-                        ],
+                        ),
+                      ],
+                    ),
+                    Padding(
+                      padding: const EdgeInsetsDirectional.only(
+                        start: 20,
+                        end: 20,
+                        bottom: 20,
                       ),
-                    const SizedBox(width: 20),
-                    Transform.translate(
-                      offset: Offset(0, isDesktop ? -18.5 : 0),
-                      child: ClipOval(
-                        child: Image.asset(
-                          'reply/avatars/avatar_0.jpg',
-                          package: _assetsPackage,
-                          height: 36,
-                          width: 36,
+                      child: Text(
+                        email.message.substring(
+                          0,
+                          email.message.indexOf(RegExp(r'[\n\n]|!|[.]'), 0) + 1,
```

Comment:
this is like voodoo to me, what's actually going on? :P

Reply:
Haha it's searching for the first occurrence of a `.` or `!` or `\n\n' and returns a substring from index 0 to the index of that first occurrence. This lets me take part of the email message and show it on the preview card. There is probably a cleaner way to do this so i'll revisit this.

Reply:
Updated the implementation!